bagit.py

```diff
@@ -1219,16 +1226,37 @@ def _parse_tags(tag_file):
         yield (tag_name, tag_value.strip())
 
 
-def _make_tag_file(bag_info_path, bag_info):
+def _make_tag_file(bag_info_path, bag_info, line_length):
     headers = sorted(bag_info.keys())
     with open_text_file(bag_info_path, "w") as f:
         for h in headers:
             values = bag_info[h]
             if not isinstance(values, list):
                 values = [values]
             for txt in values:
-                # strip CR, LF and CRLF so they don't mess up the tag file
-                txt = re.sub(r"\n|\r|(\r\n)", "", force_unicode(txt))
+                txt = force_unicode(txt)
+                if line_length > 1:
+                    # Account for colon & space written after the property name.
+                    prop_width = len(h) + 2
+                    first_break = prop_width + len(txt.split(None, 1)[0])
+                    if line_length <= first_break:
+                        # Start value on the next line.
+                        txt = '\n'.join(textwrap.wrap(txt, width=line_length,
```

Comment:
These two calls are almost identical but it takes time to confirm that. Do you think it would be worth assigning `initial_indent` first so we can have an identical textwrap call? I'm also wondering whether this block of code is now long enough that we should break it into a separate function so this would just be `for txt in values: txt = wrap_tag(h, txt, line_length)`

Reply:
Regarding the duplication: although `initial_indent` could be pulled out so that there's only one call to `wrap`, the problem is that there's a 2nd step after the call to `wrap` that depends on the same condition (see line 1257). So, the resulting alternative code would have a conditional for setting `initial_indent`, followed by the call to `wrap`, followed by an identical conditional to adjust the output of `wrap`.  I find this alternative less readable than the current code (I tried it to see), although I realize this may be a matter of personal preference. Still, what about simply moving the parameter `initial_indent` higher up in the call to `wrap`, so that it's easier to spot the difference quickly?

Regarding splitting it out to a separate function: agreed. However, I might recommend naming the function differently, if you don't object, because in this case it wouldn't necessarily wrap the tag value. So perhaps something like `_format_tag_value`?

Reply:
I went ahead and implemented the changes discussed above in the latest commit.